src/VisualStudio/Core/Impl/Options/AbstractOptionPreviewViewModel.cs

```diff
@@ -66,7 +66,10 @@ protected AbstractOptionPreviewViewModel(OptionSet options, IServiceProvider ser
 
         internal OptionSet ApplyChangedOptions(OptionSet optionSet)
         {
-            foreach (var optionKey in this.Options.GetChangedOptions(_originalOptions))
+            if (!(this.Options is IInternalOptionSet internalOptions))
+                throw new NotSupportedException();
```

Comment:
Wouldn't this be better implemented as an abstract property of the `IInternalOptionSet` type?

Reply:
Another example. Non-public method. No IVT issue. 

---
In reply to: [191925864](https://github.com/dotnet/roslyn/pull/27263#discussion_r191925864) [](ancestors = 191925864)